### PR TITLE
fix: the port name in gatekeeper service

### PR DESCRIPTION
**Pull request by GoelDeepak**
**What type of PR is this?**
Bug

**What this PR does/ why we need it**:
It fixes the port name from http to https. This is needed as it confuses istio which depend on port name of all the services in a cluster (don't be surprised ... its istio!! )

**Which issue(s) this PR fixes**:
https://jira.d2iq.com/browse/D2IQ-71798

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
No

```release-note

```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [ ] The commit message explains the changes and why are needed.
* [ ] The code builds and passes lint/style checks locally.
* [ ] The relevant subset of integration tests pass locally.
* [ ] The core changes are covered by tests.
* [ ] The documentation is updated where needed.

### CODE REVIEW

File: staging/gatekeeper/templates/service.yaml [hunk 1]
```diff
@@ -14,7 +14,7 @@ spec:
   - port: {{ .Values.service.port }}
     targetPort: http
     protocol: TCP
-    name: http
+    name: https
```
Comment (samba) [inline staging/gatekeeper/templates/service.yaml:17]: This seems inconsistent -- `targetPort: http ... name: https`. Is this intentional? Should it be the same, both corrected to `https`?

**Review by joejulian — APPROVED** [review]

File: staging/gatekeeper/templates/service.yaml (hunk 1 above)
Comment (GoelDeepak) [reply staging/gatekeeper/templates/service.yaml:17]: I am not sure. Unlike `name` which is just a hint, `targetPort` is the actual port on which service is listening. I don't know enough to change the listening port.

Comment (joejulian) [reply staging/gatekeeper/templates/service.yaml:17]: https://github.com/mesosphere/charts/blob/813c68dd6f8c8811e56ec243a74e3ae04d451dfc/staging/gatekeeper/templates/deployment.yaml#L48

Comment (joejulian) [reply staging/gatekeeper/templates/service.yaml:17]: Isn't this actually http though?

**Review by joejulian — CHANGES_REQUESTED** [review]
I linked in the comment to the line that needs change to keep the targetPort name consistent

File: staging/gatekeeper/templates/service.yaml (hunk 1 above)
Comment (GoelDeepak) [reply staging/gatekeeper/templates/service.yaml:17]: @joejulian yeah, it is `http`. However, the issue is here https://github.com/mesosphere/charts/blob/master/staging/gatekeeper/values.yaml#L39. Why that needs to be `443`?

Comment (joejulian) [reply staging/gatekeeper/templates/service.yaml:17]: If it's http, why does it have all the certificate files?
https://github.com/mesosphere/charts/blob/813c68dd6f8c8811e56ec243a74e3ae04d451dfc/staging/gatekeeper/templates/deployment.yaml#L63-L68

I didn't think webhooks were allowed to _not_ be https.

Comment (joejulian) [reply staging/gatekeeper/templates/service.yaml:17]: As an aside, the new upstream chart doesn't allow the service port to be specified in the values.

Comment (GoelDeepak) [reply staging/gatekeeper/templates/service.yaml:17]: Thats nice. I guess the right course of action in that case is to just update the chart. I'll make necessary changes in this PR.

**Comment by GoelDeepak** [comment]
Closing this in favor of upgrading the gatekeeper chart

**Pull request closed without merging**